app/authentication/token.py

```diff
@@ -36,13 +44,21 @@ async def is_admin(token: str = Depends(oauth2_scheme)) -> bool:
 
     return await is_app_admin(token, "gfw", "Unauthorized")
 
-async def is_gfwpro_admin(error_str: str, token: str = Depends(oauth2_scheme)) -> bool:
-    """Calls GFW API to authorize user.
-
-    User must be ADMIN for gfw pro app
+async def is_gfwpro_admin_for_query(dataset: str = Depends(dataset_dependency),
+                                    token: str | None = Depends(oauth2_scheme_no_auto)) -> bool:
+    """If the dataset is protected dataset, calls GFW API to authorize user by
+    requiring the user must be ADMIN for gfw-pro app.  If the dataset is not
+    protected, just returns True without any required token or authorization.
     """
-
-    return await is_app_admin(token, "gfw-pro", error_str)
+    
+    if dataset in PROTECTED_QUERY_DATASETS:
+        if token == None:
+            raise HTTPException(status_code=401, detail="Unauthorized query on a restricted dataset")
+        else:
+            await is_app_admin(cast(str, token), "gfw-pro",
```

Comment:
If is_app_admin() doesn't cause an exception because token has gfw-pro admin privileges, then we will flow to line 61 and return True there.  So, I think it is maybe redundant to add another `return True` in the else clause.  So, I won't bother changing this.